modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java

```diff
@@ -663,8 +664,13 @@ else if (task != null) {
                             // Start task execution in another thread.
                             if (sys)
                                 ctx.getSystemExecutorService().execute(taskWorker);
-                            else
-                                ctx.getExecutorService().execute(taskWorker);
+                            else {
+                                final Object executerName = map.get(EXECUTOR_NAME);
+                                if (executerName != null && executerName instanceof String)
```

Comment:
Yep, you are right we don't need class cast here because we always put String in this entry.